spring-kafka/src/test/java/org/springframework/kafka/retrytopic/BackOffPolicyProviderTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.retrytopic;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.retry.backoff.BackOffPolicy;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.backoff.ExponentialRandomBackOffPolicy;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.backoff.UniformRandomBackOffPolicy;
+
+/**
+ * @author Tomaz Fernandes
+ * @since 2.8.4
+ */
+class BackOffPolicyProviderTest {
```

Comment:
We prefer to name test classes as `*Tests`, but I guess this one could go away and its tests can migrate to the `RetryTopicConfigurationBuilderTests`. From there you can take that `backOffPolicy` from the builder using `DirectFieldAccessor`.